rfcs/0085-ZHF-on-master-and-freeze-dates.md

```diff
@@ -0,0 +1,104 @@
+was also largely responsible delaying the release until after the scheduled release date
+where basic login features in plasma [weren't restored until Oct 5th, a week after the planned release date](https://github.com/NixOS/nixpkgs/pull/99629)
+and a backlog of other plasma related issues [weren't resolved until Oct 19th, almost a month after the planned release date](https://github.com/NixOS/nixpkgs/pull/101078).
+This is not to discredit any individual, but demonstrate that certain packages
+have many dimensions of compatibility which need a longer time to stabilize
+on unstable before being included in a release.
+
+## Core changes
+
+Staging-next iterations will change to one week.
+
+Updates to `staging` which will land before branch-off will be restricted to non-breaking updates, and will need to be more thoroughly tested. The below table is a more detailed timeline, it is also sanitized of events not relating to merge criteria, staging events, or ZHF.
+
+| Weeks from Release | Events |
+| --- | --- |
+| -8 Weeks | Gnome, Plasma(YY.11) packaging begins |
+| -6 Weeks | Freeze "critical packages" breaking updates |
+| -4 Weeks | Freeze staging, allow only non-breaking updates and Desktop Manager changes |
+| -3 Weeks | (Day before ZHF) merge staging-next, prep for ZHF |
+| -3 Weeks | Begin ZHF on master branch |
+| -3 Weeks | Strong emphasis on minimizing regressions in master PRs |
+| -2 Weeks | Merge first staging-next fixes, begin second staging-next fix cycle |
+| -2 Weeks | Unfreeze "critical packages" changes to staging |
+| -2 Weeks | Unfreeze staging, allow for normal staging workflow, these changes will not be present in master before branch-off |
+| -2 Weeks | `staging-next` remains fixes only, will be merged before branch-off |
+| -1 Weeks | Merge second staging-next fix cycle |
+| -1 Weeks | Perform Branch-off, create release channels, create new beta / unstable tags |
+| -1 Weeks | ZHF transitions to "backporting" workflow |
+| -1 Weeks | Prepare for release, finish remaining issues |
+| 0 Weeks | Release! |
+| 0 Weeks | ZHF Ends |
+
+Actual release schedule may adjust dates slightly due to unforeseen events.
+
+Release team will retain the right to refine what constitutes "critical packages", as this may
+change over time, and will likely not warrant an RFC to caputre changes.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+Normal staging development is disrupted for two weeks. (1 normal iteration)
+Changes to "critical packages" may have to wait for a month to be merged into staging.
+Master development will be uninterrupted, assuming all changes don't introduce new regressions.
+
+# Alternatives
+[alternatives]: #alternatives
+
+## Maintain the status quo regarding release dates
+
+Continue with cramming staging changes right before branch-off, keeping
+a significant amount of stabilization work and risk before a release, and likely
+continuing the trend of delayed releases.
```

Comment:
```suggestion
continuing the trend of delayed releases.

## Better backport automation
A backport github action that can be would automatically create a backport of pull request. This would be also useful even if we decide to accept this pull request when a new NixOS version is released.
```

Reply:
Although I really like the idea of automated backports, I don't think it's too relevant to what the RFC is addressing. But if you want me to add it to the future work section I can, and that can be tackled at a later time.